engine/core/src/main/java/org/datacleaner/configuration/RemoteServerConfigurationImpl.java

```diff
@@ -19,20 +19,51 @@
  */
 package org.datacleaner.configuration;
 
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.net.Socket;
+import java.net.URL;
 import java.util.ArrayList;
 import java.util.Collections;
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import org.datacleaner.job.concurrent.ScheduledTaskRunner;
+import org.datacleaner.job.concurrent.TaskListener;
+import org.datacleaner.job.concurrent.TaskRunner;
+import org.datacleaner.job.tasks.Task;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /**
  * Implementation of {@link RemoteServerConfiguration}.
  */
 public class RemoteServerConfigurationImpl implements RemoteServerConfiguration {
 
+    private static final Logger logger = LoggerFactory.getLogger(RemoteServerConfigurationImpl.class);
+    private static final int TEST_CONNECTION_TIMEOUT = 15 * 1000; // [ms]
+    private static final long DELAY_MIN = 5;
```

Comment:
That would certainly also make sense, though it may make it hard to get back up if the server gets pounded when it is already unstable.

- But that is probably not a worry until we have a lot more customers, and then we'll probably need to change the architecture to better support scale-out anyway.